### ISSUE TITLE
Downloaded songs not playing

### ISSUE THREAD

**Issue by pollocluck** [jmshrv/finamp#885]
The songs I want to download play great when streaming from my server, but when I download them through the app, the song doesn't play and the stop icon flashes for one millisecond. In the download folder, the songs don't have a file type, so I'm not sure how it works with the app.

**Comment by Chaphasilor** [comment]
Hi, which version aee you using?

**Comment by pollocluck** [comment]
> Hi, which version aee you using?

Version 0.6.22 with iOS 17.6.1 and latest jellyfin server version.

**Comment by Chaphasilor** [comment]
The download system in the stable version isn't the best, and had trouble with file extensions in the past.  
Would you mind giving the beta version a try, through Testflight? You can find the link on the [release page](https://github.com/jmshrv/finamp/releases).  
After upgrading you'll want to re-download everything to get all of the inprovements. While your existing downloads are migrated of course, they will still be as broken as they were downloaded.

**Comment by pollocluck** [comment]
Yeah the downloaded songs are playing now, thanks for the suggestion! I'm hoping there's an ETA for the stable download system.

**Comment by scooterscot** [comment]
Hello,

I was having the same issue as OP and the current beta has fixed my problem. Thank you and I hope this makes it to the full version soon :)

**Comment by Chaphasilor** [comment]
That might take a while. I'd like to get most of the screens redesigned first before bringing this major update to the stable version, so that people only have to get used to it once.  
The beta is for collecting feedback before that, so that the final result makes as many people as possible happy :)

**Comment by KilianSander** [comment]
Hi, I have the same problem on iPhone (iOS 26.3.1) with Finamp 0.6.25 and Jellyfin Server 10.11.6 but just for `m4a` files:
An `m4a` file (codec: AAC; codec tag: mp4a; no AVC; LC Profile; stereo; 260 kbps; 44.1 kHz) streams fine but does not play when downloaded while downloaded `mp3` files play back properly.
The downloaded `m4a` is missing its file extension in the download folder, whereas `mp3` files have theirs.

I will try to check the `m4a` files with the TestFlight beta soon. However, I thought I should document the codec-dependent variant of the issue.
